### ISSUE TITLE
sdk 10.3.0 changed firestore webchannel to use xhr instead of fetch

### ISSUE THREAD

**Issue by atsjo** [firebase/firebase-js-sdk#7581]
### Operating System

Windows 11

### Browser Version

Chrome/Edge latest

### Firebase SDK Version

10.3.0

### Firebase SDK Product:

Firestore

### Describe your project's tooling

Angular app using esbuild builder

### Describe the problem

In sdk 10.2.0 fetch is used for firestore webchannel (I can se that from network tab in devtools, I think it's been used since 9.0), but 10.3.0 changed firestore webchannel to use xhr instead.

Overriding via undocumented useFetchStreams param in initializeFirestore doesn't work anymore either, so guessing this is a bug, and related to https://github.com/firebase/firebase-js-sdk/pull/7569. This is supposed to "fix how we enable fetch streams" to correct a bug, but seems to completely disable it. I can observe more aborted/cancelled communication in devtools with xhr, but it's still working...

### Steps and code to reproduce issue

upgrade to 10.3.0 and observe communication via devtools

**Comment by sampajano** [comment]
@atsjo Thanks for the report!

I've taken a quick look — I think with https://github.com/firebase/firebase-js-sdk/pull/7569, it is expected that all **POST (client -> server) request** will now be using XHR, but the **GET (server -> client streaming) requests** should still be using Fetch.

Is that inline with what you have observed also?

---

Disclaimer: I'm not familiar with Firestore requirements overall so i'm not sure if this is a breaking issue for anyone. I'll let members of the Firebase team comment later.

@wu-hui FYI

**Comment by atsjo** [comment]
in 10.2.0 and earlier both GET and POST for the webchannel uses fetch, but direct requests, like getCountFromServer uses xhr.
![image](https://github.com/firebase/firebase-js-sdk/assets/16601899/3d9a99f4-73dd-4acd-8aab-7bc8c4496866)

I am using persistent multitab cache if that matters...

in 10.3.0 both GET and POST for the webchannel uses xhr... 
![image](https://github.com/firebase/firebase-js-sdk/assets/16601899/c99d59fb-679c-4f55-bed6-07871187af6f)

**Comment by wu-hui** [comment]
Note the fix does what @sampajano described in https://github.com/firebase/firebase-js-sdk/issues/7581#issuecomment-1690709758 

```
all POST (client -> server) request will now be using XHR, but the GET (server -> client streaming) requests should still be using Fetch.
```

**Comment by maccman** [comment]
@wu-hui Can you elaborate on why it's bad if Firebase uses XHR versus Fetch? Is it just that the XHR API is deprecated?

**Comment by wu-hui** [comment]
Yeah, pretty much. I don't think XHR vs Fetch with requests makes much difference, it's more about we want to use newer APIs as long as it makes sense.

Using Fetch both directions with webchannel's bidi-streams is under-tested however, so we will stay on the default configuration for now, and maybe flip to that later. Again, I don't think there is much to gain here performance-wise.

**Comment by atsjo** [comment]
Tried with 10.3.1, I assume the fix is rolled out there, but its still uses xhr for everything... (both POST and GET). I also observe typically one failed GET when the app starts up (as shown in the screenshot from dev-tools above). I don't care it its using xhr of fetch, just found it strange flipping everything back to xhr after running with fetch for years, with the changelog only mentioning that it changes how it enables fetch... The undocumented useFetchStreams also stopped working, and after running with fetch for years, I would assume using xhr for everything is under-tested, as mentioned above...

**Comment by wu-hui** [comment]
The fix is not out with 10.3.1, should be the one after this.

**Comment by atsjo** [comment]
Verified canary build working as expected:-)
